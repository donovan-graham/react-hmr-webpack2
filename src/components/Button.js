import React, { PropTypes } from 'react';

// styles
import styles from './Button.css';

const Button = ({ onClick, children }) => (
  <button className={styles.btn} onClick={onClick}>{children}</button>
);

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.any,
};


export default Button;
