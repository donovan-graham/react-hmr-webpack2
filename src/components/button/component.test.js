import React from 'react';
import Button from './component';
import renderer from 'react-test-renderer';

describe('Button component', () => {
  it('default', () => {
    const tree = renderer.create(<Button>default</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('default large', () => {
    const tree = renderer.create(<Button large>default large</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('primary', () => {
    const tree = renderer.create(<Button>primary</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('primary large', () => {
    const tree = renderer.create(<Button primary large>primary large</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
