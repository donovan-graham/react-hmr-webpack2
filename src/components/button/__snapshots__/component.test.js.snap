// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component default 1`] = `
<button
  className="exZpUW"
>
  default
</button>
`;

exports[`Button component default large 1`] = `
<button
  className="hOoguq"
>
  default large
</button>
`;

exports[`Button component primary 1`] = `
<button
  className="exZpUW"
>
  primary
</button>
`;

exports[`Button component primary large 1`] = `
<button
  className="cGYEQn"
>
  primary large
</button>
`;
